src/form.js

```diff
@@ -465,7 +471,7 @@ define([
                 });
             if (meta) {
                 if (meta.dropReference(mug, property)) {
-                    this.instanceMetadata = _.without(this.instanceMetadata, meta);
+                    meta.internal = true;
```

Comment:
I don't understand this change. Also, the docstring is now outdated and should be updated if this is kept. It seems that this should not have caused an instance without a `src` to be dropped since it finds the instance by `src`. Can you explain?